src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpointUnitTest.java

```diff
@@ -803,6 +817,89 @@ public void createCaseEventNoNewCase() throws Exception {
     actions.andExpect(jsonPath("$.subCategory").doesNotExist());
   }
 
+  @Test
+  public void getCasesByPartyId() throws Exception {
+    // Given
+    PageRequest pageRequest = new PageRequest(0, 100);
+    UUID partyId = UUID.randomUUID();
+    ArgumentCaptor<Example> captor = ArgumentCaptor.forClass(Example.class);
+    when(caseRepository.findAll(captor.capture(), eq(pageRequest)))
+        .thenReturn(
+            new PageImpl<>(Collections.singletonList(Case.builder().partyId(partyId).build())));
+
+    // When
+    ResultActions actions = mockMvc.perform(getJson("/cases").param("partyId", partyId.toString()));
+
+    // Then
+    actions.andExpect(status().isOk()).andExpect(jsonPath("$[0].partyId", is(partyId.toString())));
+    Example<Case> captured = captor.getValue();
+    assertEquals(captured.getProbe().getPartyId(), partyId);
+  }
+
+  @Test
+  public void getCasesBySampleUnitId() throws Exception {
+    // Given
+    PageRequest pageRequest = new PageRequest(0, 100);
+    UUID sampleUnitId = UUID.randomUUID();
+    ArgumentCaptor<Example> captor = ArgumentCaptor.forClass(Example.class);
+    when(caseRepository.findAll(captor.capture(), eq(pageRequest)))
+        .thenReturn(
+            new PageImpl<>(
+                Collections.singletonList(Case.builder().sampleUnitId(sampleUnitId).build())));
+
+    // When
+    ResultActions actions =
+        mockMvc.perform(getJson("/cases").param("sampleUnitId", sampleUnitId.toString()));
+
+    // Then
+    actions
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].sampleUnitId", is(sampleUnitId.toString())));
+    Example<Case> captured = captor.getValue();
+    assertEquals(captured.getProbe().getSampleUnitId(), sampleUnitId);
+  }
+
+  @Test
+  public void getCasesNoParams() throws Exception {
```

Comment:
Does there need to be a test for both params present?

Reply:
@jcox-dev As far as I'm aware, party id would be null for social since information relating to a party is stored elsewhere.

Reply:
Test added